cobra/flux_analysis/gapfilling.py

```diff
@@ -2,8 +2,350 @@
 
 from __future__ import absolute_import
 
-from cobra.exceptions import DefunctError
+from sympy import Add
+from warnings import warn
 
+from optlang.interface import OPTIMAL
+from cobra.core import Model
+from cobra.util import assert_optimal, get_objective_for
 
-def growMatch():
-    raise DefunctError('growMatch', 'modelseed', 'http://modelseed.org/')
+
+class GapFiller(object):
+    """Class for performing gap filling.
+
+    This class implements gap filling based on a mixed-integer approach,
+    very similar to that described in [1]_ and the 'no-growth but growth'
+    part of [2]_ but with minor adjustments. In short, we add indicator
+    variables for using the reactions in the universal model, z_i and then
+    solve problem
+
+    minimize \sum_i c_i * z_i
+    s.t. Sv = 0
+         v_o >= t
+         lb_i <= v_i <= ub_i
+         v_i = 0 if z_i = 0
+
+    where lb, ub are the upper, lower flux bounds for reaction i, c_i is a
+    cost parameter and the objective v_o is greater than the lower bound t.
+    The default costs are 1 for reactions from the universal model, 100 for
+    exchange (uptake) reactions added and 1 for added demand reactions.
+
+    Note that this is a mixed-integer linear program and as such will
+    expensive to solve for large models. Consider using CORDA instead _[3,4].
+
+    Parameters
+    ----------
+    model : cobra.Model
+        The model to perform gap filling on.
+    objective : model.interface.Objective
```

Comment:
Now SMILEY and growMatch still do what they should but the new implementation has less mess